das/daser.go

```diff
@@ -138,8 +141,11 @@ func (d *DASer) sample(ctx context.Context, sub header.Subscription, checkpoint
 
 		startTime := time.Now()
 
-		err = d.da.SharesAvailable(ctx, h.DAH)
+		ctxWithSpan, span := otel.Tracer(tracerName).Start(ctx, "sample")
+
+		err = d.da.SharesAvailable(ctxWithSpan, h.DAH)
 		if err != nil {
+			span.End()
```

Comment:
nit: do defer `span.End()`. This would give the same result, and the span will end on any success or error return.

Reply:
We should record errors so that the operation does not look like a success. https://opentelemetry.uptrace.dev/guide/go-tracing.html#recording-errors